src/System.Text.Json/tests/Serialization/Array.ReadTests.cs

```diff
@@ -3,14 +3,74 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
-using System.Collections.Immutable;
-using System.Reflection;
 using Xunit;
 
 namespace System.Text.Json.Serialization.Tests
 {
     public static partial class ArrayTests
     {
+        [Fact]
+        public static void ReadObjectArray()
+        {
+            string data =
+                "[" +
+                SimpleTestClass.s_json +
+                "," +
+                SimpleTestClass.s_json +
+                "]";
+
+            SimpleTestClass[] i = JsonSerializer.Parse<SimpleTestClass[]>(Encoding.UTF8.GetBytes(data));
+
+            i[0].Verify();
+            i[1].Verify();
+        }
+
+        [Fact]
+        public static void ReadEmptyObjectArray()
+        {
+            SimpleTestClass[] data = JsonSerializer.Parse<SimpleTestClass[]>("[{}]");
+            Assert.Equal(1, data.Length);
+            Assert.NotNull(data[0]);
+        }
+
+        [Fact]
+        public static void ReadPrimitiveJagged2dArray()
+        {
+            int[][] i = JsonSerializer.Parse<int[][]>(Encoding.UTF8.GetBytes(@"[[1,2],[3,4]]"));
```

Comment:
Does int[][] within a container work?

For example, from https://github.com/dotnet/corefx/issues/36889#issuecomment-493293595
```C#
class Container
{
    public int[][] Matrix { get; set; }
}

var json = "{\"Matrix\": [[1, 2], [3, 4]]}";
Console.WriteLine(JsonSerializer.Parse<Container>(json));
```

Reply:
Yep, it does:
```C#
Console.WriteLine(JsonSerializer.Parse<Container>(json).Matrix[1][1]); // 4
```

Reply:
See also https://github.com/dotnet/corefx/pull/37867/files#diff-09b2ce57590123ca00b2737282792272R52